File: well-core/src/main/java/com/wellnetworks/wellcore/java/domain/BaseEntity.java
package com.wellnetworks.wellcore.java.domain;

import jakarta.persistence.Column;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.MappedSuperclass;
import lombok.Getter;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedBy;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@EntityListeners(AuditingEntityListener.class)
@Getter
@MappedSuperclass
public class BaseEntity {

    @CreatedDate
    @Column(updatable = false)
    private LocalDateTime createdAt; //yyyy-mm-dd hh:mm:ss

    @LastModifiedDate
    private LocalDateTime updatedAt; //yyyy-mm-dd hh:mm:ss

    @Column(name = "writer", updatable = false) //작성자
    @CreatedBy
    private String writer;

    @Column(name = "updater") //수정자
    @LastModifiedBy
    private String updater;
}

File: well-core/src/main/kotlin/com/wellnetworks/wellcore/domain/BaseEntity.kt
package com.wellnetworks.wellcore.domain

import jakarta.persistence.Column
import jakarta.persistence.MappedSuperclass
import java.time.ZonedDateTime

@MappedSuperclass
// MappedSuperclass : 여러 엔티티 클래스에서 중복되는 코드를 줄이고 코드 재사용성을 높이기 위해 사용
open class BaseEntity {
    @Column(name = "mod_dt", nullable = true)
    open var modifyDatetime: ZonedDateTime? = ZonedDateTime.now()

    @Column(name = "reg_dt", nullable = true)
    open var registerDatetime: ZonedDateTime? = ZonedDateTime.now()
}